packages/typescript-estree/src/create-program/createWatchProgram.ts

```diff
@@ -246,8 +246,7 @@ function createWatchProgram(
   watchCompilerHost.readFile = (filePathIn, encoding): string | undefined => {
     const filePath = getCanonicalFileName(filePathIn);
     const fileContent =
-      path.normalize(filePath) ===
-      path.normalize(currentLintOperationState.filePath)
+      filePath === currentLintOperationState.filePath
```

Comment:
I noticed here that this is normalising an already normalised path:
Both `filePath` and `currentLintOperationState.filePath` are `CanonicalPath`